ui/app/AppLayouts/Profile/views/PasswordView.qml

```diff
@@ -0,0 +1,365 @@
+            anchors.verticalCenter: parent.verticalCenter
+            anchors.right: parent.right
+            anchors.rightMargin: 16
+            width: 24
+            height: 24
+            icon.name: currentPswInput.showPassword ? "hide" : "show"
+            icon.color: Theme.palette.baseColor1
+
+            onClicked: currentPswInput.showPassword = !currentPswInput.showPassword
+        }
+    }
+
+    Column {
+        spacing: Style.current.padding / 2
+        z: root.zFront
+        width: parent.width
+
+        // TODO replace with StatusInput as soon as it supports password
+        Input {
+            id: newPswInput
+
+            property bool showPassword
+
+            width: parent.width
+            placeholderText: qsTr("New password")
+            textField.echoMode: showPassword ? TextInput.Normal : TextInput.Password
+            textField.validator: d.validator
+            keepHeight: true
+            textField.rightPadding: showHideNewIcon.width + showHideNewIcon.anchors.rightMargin + Style.current.padding / 2
+
+            onTextChanged: {
+                // Update password checkers
+                d.containsLower = d.lowerCaseValidator(text)
+                d.containsUpper = d.upperCaseValidator(text)
+                d.containsNumbers = d.numbersValidator(text)
+                d.containsSymbols = d.symbolsValidator(text)
+
+                // Update strength indicator:
+                strengthInditactor.strength = d.getPswStrength()
+            }
+
+            StatusFlatRoundButton {
+                id: showHideNewIcon
+                visible: newPswInput.text !== ""
+                anchors.verticalCenter: parent.verticalCenter
+                anchors.right: parent.right
+                anchors.rightMargin: 16
+                width: 24
+                height: 24
+                icon.name: newPswInput.showPassword ? "hide" : "show"
+                icon.color: Theme.palette.baseColor1
+
+                onClicked: newPswInput.showPassword = !newPswInput.showPassword
+            }
+        }
+
+        StatusPasswordStrengthIndicator {
+            id: strengthInditactor
+            width: parent.width
+            anchors.horizontalCenter: parent.horizontalCenter
+            value: newPswInput.text.length > root.minPswLen ? root.minPswLen : newPswInput.text.length
+            from: 0
+            to: root.minPswLen
+            labelVeryWeak: qsTr("Very weak")
+            labelWeak: qsTr("Weak")
+            labelSoso: qsTr("So-so")
+            labelGood: qsTr("Good")
+            labelGreat: qsTr("Great")
+        }
+    }
+
+    StatusBaseText {
+        id: strengthenTxt
+        anchors.horizontalCenter: parent.horizontalCenter
+        text: root.strengthenText
+        font.pixelSize: 12
+        color: Theme.palette.baseColor1
+    }
+
+    Row {
+        spacing: Style.current.padding
+        anchors.horizontalCenter: parent.horizontalCenter
+
```

Comment:
Just wondering, would all text in one single line separated with spaces look also ok? If not, is there any strong reason besides the spacing to instantiate 4 different Text components inside a Row?

Reply:
no it's not! so let's leave it as it :)